Add documentation to ecoforest integration (#28953)
home-assistant/home-assistant.github.io: source/_integrations/ecoforest.markdown

---
title: Ecoforest
description: Instructions on how to integrate Ecoforest fireplaces with Home Assistant.
ha_category:
  - Climate
ha_release: 2023.10
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@pjanuario'
ha_domain: ecoforest
ha_platforms:
  - sensor
ha_integration_type: integration
---

> The Ecoforest integration allows monitoring and control of local [Ecoforest](https://ecoforest.com) fireplaces in Home Assistant. These fireplaces are manufactured in Spain.

Reviewer: Why does it matter where they are manufactured for this introduction text?

Author: Changes requested are here https://github.com/home-assistant/home-assistant.io/pull/28975

There is currently support for the following device platforms within Home Assistant:

- [Sensor](#sensor)

> {% include integrations/config_flow.md discovery=False %}

Reviewer 2: ```suggestion
{% include integrations/config_flow.md %}
```

<div class='note'>

> Devices must first be connected to Wi-Fi or LAN using the app or website provided by the device manufacturer. It is not possible to connect devices to Wi-Fi using this integration.

Reviewer: This sentence if odd / confusing:

`Devices must first be connected to Wi-Fi or LAN`

Which is next followed by:

`It is not possible to connect devices to Wi-Fi using this integration.`

So the first sentence is not correct / true?

Author: Changes requested are here https://github.com/home-assistant/home-assistant.io/pull/28975

</div>

### Credentials

To configure the Ecoforest integration you will need to enter your Ecoforest credentials which are the same ones you would use with the manufacturer app. The image below shows how to obtain the credentials:

- Username: use the serial number of the device as identified by 1.
- Password: use the first 8 characters of the wifi password as identified by 4.

![Ecoforest Credentials](/images/integrations/ecoforest/credentials.png)

> 

Reviewer: @joostlek @pjanuario It doesn't make much sense to have install instructions above how to get the details needed to install. This is a prerequisite...

Author: I was thinking in the UX for less technical users, I can remove if we fill it shouldn't be there.

Author: Changes requested are here https://github.com/home-assistant/home-assistant.io/pull/28975

---
title: Ecoforest
description: Instructions on how to integrate Ecoforest fireplaces with Home Assistant.
ha_category:
  - Climate
ha_release: 2023.10
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@pjanuario'
ha_domain: ecoforest
ha_platforms:
  - sensor
ha_integration_type: integration
---

The Ecoforest integration allows monitoring and control of local [Ecoforest](https://ecoforest.com) fireplaces in Home Assistant. These fireplaces are manufactured in Spain.

There is currently support for the following device platforms within Home Assistant:

- [Sensor](#sensor)

{% include integrations/config_flow.md discovery=False %}

<div class='note'>
Devices must first be connected to Wi-Fi or LAN using the app or website provided by the device manufacturer. It is not possible to connect devices to Wi-Fi using this integration.
</div>

### Credentials

> To configure the Ecoforest integration you will need to enter your Ecoforest credentials which are the same ones you would use with the manufacturer app. The image bellow shows how to obtain the credentials:

Reviewer 2: ```suggestion
To configure the Ecoforest integration you will need to enter your Ecoforest credentials which are the same ones you would use with the manufacturer app. The image below shows how to obtain the credentials:
```

- Username: use the serial number of the device as identified by 1.
- Password: use the first 8 characters of the wifi password as identified by 4.

![Ecoforest Credentials](/images/integrations/ecoforest/credentials.png)

## Supported models

Any Ecoforest device working with [Ecoforest Home](https://ecoforesthome.com/) should be supported, this integration is confirmed to support:

- Ecoforest Cordoba Glass (using firmware version `30Abr19_v2z`)

## Sensor

The Ecoforest integration exposes a sensor for the current ambient temperature

> 
> - Temperature: Current ambient temperature as read by the device probe.

Reviewer 2: ```suggestion
```
When you add more sensors we can add this list back, but now it just looks like a sad list

---
title: Ecoforest
description: Instructions on how to integrate Ecoforest fireplaces with Home Assistant.
ha_category:
  - Climate

> ha_release: 2022.9

Reviewer 2: ```suggestion
ha_release: 2023.10
```

ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@pjanuario'
ha_domain: ecoforest
ha_platforms:
  - sensor
ha_integration_type: integration
---

The Ecoforest integration allows monitoring and control of local [Ecoforest](https://ecoforest.com) fireplaces in Home Assistant. These fireplaces are manufactured in Spain.

There is currently support for the following device platforms within Home Assistant:

- [Sensor](#sensor)

> {% include integrations/config_flow.md discovery=False %}

Reviewer 2: ```suggestion
{% include integrations/config_flow.md %}
```
Isn't that the default?

Author: No, if I don't include the discovery flag to false it mentions in the text about device discovery wich is not true for this one.

<div class='note'>

> Devices must first be bound to WiFi or LAN using the app or website provided by the device manufacturer. It is not possible to connect devices to WiFi using this integration.

Reviewer 2: ```suggestion
Devices must first be connected to the internet using the app or website provided by the device manufacturer. It is not possible to connect devices to the internet using this integration.
```

Author: Well, in this case it is local device pooling so no internet access is needed. My devices don't have internet access just local network.

Reviewer 2: right, in that case WiFi -> Wi-Fi

</div>

### Credentials

To configure the Ecoforest integration you will need to enter your Ecoforest credentials which are the same ones you would use with the manufacturer app. The image bellow shows how to obtain the credentials:

- Username: use the serial number of the device as identified by 1.
- Password: use the first 8 characters of the wifi password as identified by 4.

![Ecoforest Credentials](/images/integrations/ecoforest/credentials.png)

## Supported models

Any Ecoforest device working with [Ecoforest Home](https://ecoforesthome.com/) should be supported, this integration is confirmed to support:

- Ecoforest Cordoba Glass (using firmware version `30Abr19_v2z`)

## Sensor

> Ecoforest devices expose multiple sensors to monitor various features:

Reviewer 2: ```suggestion
The Ecoforest integration exposes a sensor for the current ambient temperature
```

Reviewer 2: I would keep it like this since it's a cut down version

Author: It's just that the other sensors are already ready to get in after the initial one 😄 
So I left it correct but ready to add it.

> - Room Temperature: Current ambient temperature as read by the device probe.

Reviewer 2: ```suggestion
- Temperature: Current ambient temperature as read by the device probe.
```

Author: You bit me on that, I was first finishing the core 😄

## Troubleshooting

If you're trying to track down issues with this integration, set up logging for it.

### Debugging

To turn on debug logging in the integration configuration view or modify your `configuration.yaml` file in the `/config` directory and add the following:

```yaml
# Example configuration.yaml with logging for Ecoforest
logger:
  default: warning
  logs:
    homeassistant.components.ecoforest: debug

> ```

Reviewer 2: These are part of usual HA tasks and can be removed
